funsor/cnf.py

```diff
@@ -220,14 +225,71 @@ def eager_contraction_to_binary(red_op, bin_op, reduced_vars, lhs, rhs):
     return result
 
 
+@eager.register(Contraction, ops.AddOp, ops.MulOp, frozenset, Tensor, Tensor)
```

Comment:
Add a pattern for `ops.MaxOp`/`ops.AddOp` that uses the `torch_map` backend?  I'm not sure that does anything smarter but it would keep execution behavior more consistent.